yt/yt/server/lib/chunk_pools/job_size_tracker.cpp

```diff
@@ -143,6 +147,26 @@ class TJobSizeTracker
             return;
         }
 
+        if (Options_.LimitProgressionRatio && LimitProgressionLength_ < Options_.LimitProgressionLength) {
```

Comment:
I don't understand what is going on here, but if it works as expected, that's fine

Reply:
The class itself is quite complicated, yeah, it took me a while. You should double check the general intent and whether I computed parameters correctly. 
I spent quite some time debugging errors around this and idk if there are more mistakes.